app/models/provisioning_template.rb

```diff
@@ -111,50 +111,52 @@ def self.find_template(opts = {})
   end
 
   def self.build_pxe_default(renderer)
-    if (proxies = SmartProxy.with_features("TFTP")).empty?
-      error_msg = _("No TFTP proxies defined, can't continue")
-    end
-
-    if (default_template = ProvisioningTemplate.find_by_name("PXELinux global default")).nil?
-      error_msg = _("Could not find a Configuration Template with the name \"PXELinux global default\", please create one.")
-    end
-
-    if error_msg.empty?
-      begin
-        @profiles = pxe_default_combos
-        allowed_helpers = Foreman::Renderer::ALLOWED_GENERIC_HELPERS + [ :default_template_url ]
-        menu = renderer.render_safe(default_template.template, allowed_helpers, :profiles => @profiles)
-      rescue => e
-        error_msg = _("failed to process template: %s" % e)
-      end
-    end
-
-    return [422, error_msg] unless error_msg.empty?
+    return [200, _("No TFTP proxies defined, can't continue")] if (proxies = SmartProxy.with_features("TFTP")).empty?
 
     error_msgs = []
-    proxies.each do |proxy|
-      begin
-        tftp = ProxyAPI::TFTP.new(:url => proxy.url)
-        tftp.create_default({:menu => menu})
-
-        @profiles.each do |combo|
-          combo[:hostgroup].operatingsystem.pxe_files(combo[:hostgroup].medium, combo[:hostgroup].architecture).each do |bootfile_info|
-            for prefix, path in bootfile_info do
-              tftp.fetch_boot_file(:prefix => prefix.to_s, :path => path)
-            end
+    TemplateKind::PXE.each do |kind|
+      default_template_name = "#{kind} global default"
+      if (default_template = ProvisioningTemplate.find_by_name(default_template_name)).nil?
+        error_msg = _("Could not find a Configuration Template with the name \"%s global default\", please create one.") % kind
+      else
+        begin
+          @profiles = pxe_default_combos
+          allowed_helpers = Foreman::Renderer::ALLOWED_GENERIC_HELPERS + [ :default_template_url ]
+          menu = renderer.render_safe(default_template.template, allowed_helpers, :profiles => @profiles)
+        rescue => exception
```

Comment:
Thinking out loud.. we should start raising errors in the rendering method, then rescuing them when we call it, instead of rescuing everything when we call it like now